adapters/undertone/undertone.go

```diff
@@ -0,0 +1,209 @@
+package undertone
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	"strconv"
+)
+
+const adapterId = 4
+const adapterVersion = "1.0.0"
+
+type adapter struct {
+	endpoint string
+}
+
+type BidRequestExt struct {
+	Prebid *openrtb_ext.ExtRequestPrebid `json:"prebid"`
+}
+
+type undertoneBidderParams struct {
+	Id      int    `json:"id"`
+	Version string `json:"version"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	imps, publisherId, errs := getImpsAndPublisherId(request)
+	if len(imps) == 0 {
+		return nil, errs
+	}
+
+	reqCopy := *request
+	reqCopy.Imp = imps
+
+	populateSiteApp(&reqCopy, publisherId, request.Site, request.App)
+	e := populateBidReqExt(&reqCopy)
+	if e != nil {
+		errs = append(errs, e)
+	}
+
+	requestJSON, err := json.Marshal(reqCopy)
+	if err != nil {
+		return nil, []error{err}
```

Comment:
`[]error{err}` shadows the previously appended errors from line 50. Preferred way will be to append and return error from line 55

```
requestJSON, err := json.Marshal(reqCopy)
if err != nil {
		return nil, append(errs, e)
}

```